src/core/diags/factory_diags.cpp

```diff
@@ -559,7 +559,7 @@ void Diags::TransmitPacket(void)
     }
 
     mDiagSendOn = true;
-    IgnoreError(Get<Radio>().Transmit(*static_cast<Mac::TxFrame *>(mTxPacket)));
+    return Get<Radio>().Transmit(*static_cast<Mac::TxFrame *>(mTxPacket));
```

Comment:
Could we output error message in the function `TransmitPacket()` to avoid duplicated code.
```c++
   error = Get<Radio>().Transmit(*static_cast<Mac::TxFrame *>(mTxPacket);
   if (error != kErrorNone)
   {
      Output("Packet transmission failed, error: %s\r\n", ErrorToString(error));
   }
```